homeassistant/components/vicare/__init__.py

```diff
@@ -85,20 +65,28 @@ def setup_vicare_api(hass: HomeAssistant, entry: ConfigEntry) -> None:
     """Set up PyVicare API."""
     vicare_api = vicare_login(hass, entry.data)
 
-    for device in vicare_api.devices:
-        _LOGGER.info(
+    device_config_list = remove_unsupported_devices(vicare_api.devices)
+    if number_of_devices := len(device_config_list) > 1:
+        cache_duration = max(
+            DEFAULT_CACHE_DURATION, DEFAULT_CACHE_DURATION * number_of_devices
+        )
+        _LOGGER.debug(
+            "Found %s devices, adjusting cache duration to %s",
+            number_of_devices,
+            cache_duration,
+        )
+        vicare_api = vicare_login(hass, entry.data, cache_duration)
+        device_config_list = remove_unsupported_devices(vicare_api.devices)
```

Comment:
why again?

Reply:
I don't get it...
Please explain why the cache is device-dependent and what the cache does.

Reply:
Each device has an api object stored on it which talkes to the endpoint (`.../installations/<installation-id>/gateways/<gateway-serial>/device/<device-id>/features`) that provides all sensor readings for a device. This response is by default cached for 60 seconds and invalidated after that time or if a property is written to the api (setTemp, activatePreset, etc.).  
There is currently no way to adjust the cache time after creation.

Reply:
@edenhaus is there something open to discuss?

Reply:
But I don't get why the cache duration is 60 seconds multiply the number of devices? Why do we need to increase it if we have more devices? Can the cache not still be 60s?

Reply:
The cached services is calling the /device/features endpoint which will return all datapoints with its values for the device. For getting the datapoints for another device, we need to make another api call. 

So each device has its own cached service object. When we have a cache duration of 60 seconds we will get 1440 api calls a day and stay within the limit of 1500. If we have two devices that update every 60 seconds we will get 2880 calls and breach the limit.

Reply:
> We can also discuss on discord.

Sure. My username is `edenhaus`